cpp/src/arrow/compute/expression.cc

```diff
@@ -220,6 +220,7 @@ void PrintTo(const Expression& expr, std::ostream* os) {
 
 bool Expression::Equals(const Expression& other) const {
   if (Identical(*this, other)) return true;
+  if (impl_ == nullptr || other.impl_ == nullptr) return false;
```

Comment:
```suggestion

  if (impl_ == nullptr || other.impl_ == nullptr) return false;
```

Reply:
Just for complying with the existing style: each returning branch is separated by a blank line.